### ISSUE TITLE
Baskup 3 MAS

### ISSUE THREAD

**Issue by syberknight** [PeterKaminski09/baskup#36]
curious about the progress in getting version 3 in to the Mac App Store (per your website info)?

and, will that app provide a nicely formatted PDF export option (with attachments thumbs inline)?

thanks :-)

**Comment by armandhammer** [comment]
Yeah it seems like this project is abandoned.

**Comment by PeterKaminski09** [comment]
Nope, not abandoned. Just not my focus right now. I’ll get to it at some point.

**Comment by armandhammer** [comment]
Thanks. I would pay money for this. I already paid for a similar app called Chatology, but it doesn't work since it only pulls messages from the Chat Archive files (flat text) and not the Chat.db -- unfortunately this means that Spotlight can't find all of my chats (even if I search directly in Messages.app) and they don't show up in Chatology, because Spotlight only indexes the Chat Archive files. The developer of Chatology wasn't able to give me a solution. This seems to be a problem since we now have iCloud Messages, which only downloads the chat.db to the Mac when you set up a new computer. And there seems to be no reliable way to force Mac OS (or Messages.app) to generate Chat Archive files for every conversation, or even the entire history of a single conversation. It's super frustrating. I feel like what you've done so far could be turned into a really nice solution for archiving and searching my entire chat history on my Mac.

**Comment by PeterKaminski09** [comment]
Gotcha—I know there is a decent amount of support for this project. I recently got my other side project onto the App Store, and I was kicking around the idea of dusting this off and bringing it up to date for the newest OS. 

It’s been about 2 years since I’ve worked on the Mac app, and so I’m not sure how the landscape has changed for processing the chat.db file. I’d love to get it onto the App Store, so I’ll fire it up and see how much work is required to do that. 🙂

**Comment by tagatac** [comment]
@bengarland, you could pledge for a MAS version: https://www.bountysource.com/ to motivate @PeterKaminski09 monetarily 🙂

**Comment by tagatac** [comment]
@PeterKaminski09, I noted as @programsam did in #26 that I can't find your Mac OS dmg build code. Are you planning to keep that closed source? If so, will the Mac version diverge in functionality from the code you have open sourced in this project? Has it already?

**Comment by PeterKaminski09** [comment]
The macOS code uses the SQLite.swift library to interact with the database. While the queries themselves are not significantly different from the ones in the script, they are a little bit different as they include some more enhanced functionality.

Good news people, I just started developing on the mac app again after 3 years. It'll take me a bit to convert it up to the latest version of Swift and do some general code clean up, but I'm hoping to have either a new .DMG or the first app store release soon (~1 month). If I put the code onto the app store, I will probably charge a small fee to download the software so I will be incentivized to maintain it more regularly. If not, I will push the code to an open source repo for people to share and improve.

**Comment by tagatac** [comment]
Sounds like a great plan, thanks for the update!

**Comment by armandhammer** [comment]
Yeah man you should definitely charge like $10 for this. People who seek out an app like this have a real need for it. That’s worth money.

**Comment by krisraeo** [comment]
Hey Peter,
Here to let you know that I’m one of those people who could really use something like Baskup. I’m looking for a way to export and archive entire (years) worth of chat/text history with several contacts from the chat.db. I’m sure I could figure out how to run your script, but having it all presented in a nice App UX sure would make life easier. I’d be your first purchase.

Looking forward to it getting on the App Store.

**Comment by cgraymetamedia** [comment]
Ditto all the demand sentiment. Since I'm been isolated this last year, a lot of my interactions have been via Imessages, and I'd like a way to export them into formatted documents. An app would save so much time...

**Comment by cgraymetamedia** [comment]
IMazing has an excellent Messages export feature, easier to use and certainly more configurable than this script. Well worth checking out.

**Comment by syberknight** [comment]
@cgraymetamedia  last i looked, iMazing could not access these messages files in other directories (like from an external drive's manual backup). i think Baskup has that ability (as far as i know - of course have not been able to try it, just hoping... still...).

**Comment by cgraymetamedia** [comment]
iMazing accesses an iOS device backup to export conversations to PDF, text, CSV. Also exports attachments. It gives you access to all the db row metadata in the CSV export, and much more granular controls than Baskup.
     https://imazing.com/transfer-iphone-text-messages-to-computer
I don't see way to for iMazing to access from just a DB backup, but it might be possible if you can somehow manually edit the iOS backup file and insert the DB files...
I have no affiliation with Baskup or iMazing. There may be other options. Baskup is a good example of directly accessing this data from a bash script.

**Comment by programsam** [comment]
@cgraymetamedia do you have to pay for iMazing? https://imazing.com/store What features are limited if you don’t? Does it time out or expire after a trial?

**Comment by tagatac** [comment]
If anyone is interested in a Go version that fixes some of the bugs and is installable with homebrew, I made https://github.com/tagatac/bagoup last year after several PRs in this project went ignored for a while. I'm planning to add PDF output format, and more usage would probably motivate me. It is still a CLI though; I'm not planning to add a GUI.
